compiler/fir/analysis-tests/testData/resolve/inference/coercionToUnitWithEarlyReturn.kt

```diff
@@ -15,5 +15,5 @@ fun main(x: A?) {
     }
 
     // lambda has a type (() -> Unit?)
-    foo(lambda)
+    <!INAPPLICABLE_CANDIDATE{LT}!><!INAPPLICABLE_CANDIDATE{PSI}!>foo<!>(lambda)<!>
```

Comment:
See my comment here: https://youtrack.jetbrains.com/issue/KT-39075#focus=Comments-27-4670446.0-0

Now it's `() -> Unit?` type as mentioned at the code comment above. Eventually, we need to handle the early return inside the lambda properly, but it would be a different topic.

Reply:
We need a special handling (a.k.a. coercion-to-Unit (for the last expression)) when a lambda has early returns. 3rd commit adds that.